server/catgenome/src/main/java/com/epam/catgenome/manager/blast/BlastTaskManager.java

```diff
@@ -0,0 +1,167 @@
+import com.epam.catgenome.exception.BlastRequestException;
+import com.epam.catgenome.manager.AuthManager;
+import com.epam.catgenome.util.db.Filter;
+import com.epam.catgenome.util.db.QueryParameters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.lang.Thread.sleep;
+
+@Service
+public class BlastTaskManager {
+    private static final int CHECK_TASK_STATUS_PERIOD_MS = 10000;
+    @Autowired
+    private BlastTaskDao blastTaskDao;
+    @Autowired
+    private BlastRequestManager blastRequestManager;
+    @Autowired
+    private AuthManager authManager;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public Task load(Long taskId) {
+        Task task = blastTaskDao.loadTaskById(taskId);
+        Assert.notNull(task, MessageHelper.getMessage(MessagesConstants.ERROR_TASK_NOT_FOUND, taskId));
+
+        loadTaskOrganisms(task);
+        loadTaskParameters(task);
+
+        return task;
+    }
+
+    private void loadTaskOrganisms(Task task) {
+        List<Organism> organismList = blastTaskDao.loadOrganisms(task.getId());
+        List<Long> organisms = organismList.stream().map(Organism::getOrganism).collect(Collectors.toList());
+        task.setOrganisms(organisms);
+    }
+
+    private void loadTaskParameters(Task task) {
+        List<TaskParameter> parametersList = blastTaskDao.loadTaskParameters(task.getId());
+        Map<String, String> parameters = parametersList.stream().collect(Collectors.toMap(TaskParameter::getParameter, TaskParameter::getValue));
+        task.setParameters(parameters);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public Task create(final Task task) throws BlastRequestException {
+        Request request = new Request();
+        request.setAlgorithm(task.getAlgorithm());
+        request.setBlastTool("BLAST_TOOL");
+        request.setDbName(task.getDatabase());
+        request.setQuery(task.getQuery());
+        request.setOptions(task.getOptions());
+        request.setTaxIds(task.getOrganisms());
+        request.setExcludedTaxIds(task.getExcludedOrganisms());
+        RequestInfo requestInfo = blastRequestManager.createTask(request);
+        task.setId((long) requestInfo.getPayload().getRequestId());
+        task.setStatus(TaskStatus.valueOf(requestInfo.getPayload().getStatus()));
+        task.setOwner(authManager.getAuthorizedUser());
+        blastTaskDao.saveTask(task);
+        blastTaskDao.saveOrganisms(task.getId(), task.getOrganisms());
+        blastTaskDao.saveExclOrganisms(task.getId(), task.getExcludedOrganisms());
+        blastTaskDao.saveTaskParameters(task.getId(), task.getParameters());
+        new Thread(() -> {
```

Comment:
Let's use `@Scheduled` method (maybe declared in a separate class) which will load all active tasks and update its status. It will reduce number of required threads and allow to handle server restarts.